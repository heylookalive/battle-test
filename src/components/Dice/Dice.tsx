import React from 'react';
// Dice styling via https://dev.to/ekeijl/creating-dice-using-css-grid-j4
import './Dice.css';

type DiceProps = {
  value: number;
};

export default function Dice({ value }: DiceProps) {
  return (
    <div className="face">
      {Array(value)
        .fill(0)
        .map((item, key) => (
          <span className="pip" key={key} />
        ))}
    </div>
  );
}
