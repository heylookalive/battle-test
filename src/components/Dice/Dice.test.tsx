import { render } from '@testing-library/react';
import React from 'react';
import Dice from './Dice';

describe('<Dice />', () => {
  it('renders as expected', () => {
    const { container } = render(<Dice value={6} />);
    expect(container).toMatchSnapshot();
  });
});
