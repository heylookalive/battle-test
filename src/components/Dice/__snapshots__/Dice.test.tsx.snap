// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dice /> renders as expected 1`] = `
<div>
  <div
    class="face"
  >
    <span
      class="pip"
    />
    <span
      class="pip"
    />
    <span
      class="pip"
    />
    <span
      class="pip"
    />
    <span
      class="pip"
    />
    <span
      class="pip"
    />
  </div>
</div>
`;
